pylearn2/linear/tests/test_cudnn.py

```diff
@@ -0,0 +1,151 @@
+"""
+Tests for the Cudnn code
+"""
+
+import theano
+from theano import tensor
+from theano.sandbox.cuda.dnn import dnn_available
+
+from pylearn2.linear.conv2d import Conv2D
+from pylearn2.linear.cudnn2d import Cudnn2D, make_random_conv2D
+from pylearn2.space import Conv2DSpace
+from pylearn2.utils import sharedX
+from pylearn2.testing.skip import skip_if_no_gpu
+
+import unittest
+from nose.plugins.skip import SkipTest
+
+import numpy as np
+
+
+class TestCudnn(unittest.TestCase):
+    """
+    Tests for the Cudnn code
+    """
+    def setUp(self):
+        """
+        Set up a test image and filter to re-use
+        """
+        skip_if_no_gpu()
+        if not dnn_available():
+            raise SkipTest('Skipping tests cause cudnn is not available')
+        self.orig_floatX = theano.config.floatX
+        theano.config.floatX = 'float32'
+        self.image = np.random.rand(1, 1, 3, 3).astype(theano.config.floatX)
+        self.image_tensor = tensor.tensor4()
+        self.input_space = Conv2DSpace((3, 3), 1, axes=('b', 'c', 0, 1))
+        self.filters_values = np.ones(
+            (1, 1, 2, 2), dtype=theano.config.floatX
+        )
+        self.filters = sharedX(self.filters_values, name='filters')
+        self.batch_size = 1
+
+        self.cudnn2d = Cudnn2D(self.filters, self.batch_size, self.input_space)
+
+    def tearDown(self):
```

Comment:
Yep, fixed that. Apparently the latest pylint installation on ubuntu broke syntastic :/
I am pushing frequently to find out syntax/pep8 errors with Travis, don't bother pointing them out. I will check Travis logs on my own and I'll comment back when I think I am done, so you don't waste your time on stupid errors like this. :)